qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -0,0 +1,477 @@
+                coefficients of the observable.
+            callback (Callable[[int, np.ndarray, float, dict[str, Any]], None] | None):
+                a callback that can access the intermediate data
+                during the optimization. Four parameter values are passed to the callback as
+                follows during each evaluation by the optimizer: the evaluation count,
+                the optimizer parameters for the ansatz, the estimated value, the estimation
+                metadata, and the current step.
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        fidelity: BaseStateFidelity,
+        ansatz: QuantumCircuit,
+        optimizer: Optimizer | Minimizer,
+        *,
+        k: int = 2,
+        betas: Sequence[float] | None = None,
+        initial_point: Sequence[float] | None = None,
+        callback: Callable[[int, np.ndarray, float, dict[str, Any]], None] | None = None,
+    ) -> None:
+        """
+
+        Args:
+            estimator: The estimator primitive.
+            fidelity: The fidelity class using primitives.
+            ansatz: A parameterized circuit used as ansatz for the wave function.
+            optimizer: A classical optimizer. Can either be a Qiskit optimizer or a callable
+                that takes an array as input and returns a Qiskit or SciPy optimization result.
+            k: the number of eigenvalues to return. Returns the lowest k eigenvalues.
+            betas: beta parameters in the VQD paper.
+                Should have length k - 1, with k the number of excited states.
+                These hyperparameters balance the contribution of each overlap term to the cost
+                function and have a default value computed as the mean square sum of the
+                coefficients of the observable.
+            initial_point: an optional initial point (i.e. initial parameter values)
+                for the optimizer. If ``None`` then VQD will look to the ansatz for a preferred
+                point and if not will simply compute a random one.
+            callback: a callback that can access the intermediate data
+                during the optimization. Four parameter values are passed to the callback as
+                follows during each evaluation by the optimizer: the evaluation count,
+                the optimizer parameters for the ansatz, the estimated value,
+                the estimation metadata, and the current step.
+        """
+        super().__init__()
+
+        self.estimator = estimator
+        self.fidelity = fidelity
+        self.ansatz = ansatz
+        self.optimizer = optimizer
+        self.k = k
+        self.betas = betas
+        # this has to go via getters and setters due to the VariationalAlgorithm interface
+        self.initial_point = initial_point
+        self.callback = callback
+
+        self._eval_time = None
```

Comment:
I don't think _eval_time instance var is ever used?

Reply:
true